### DISCUSSION TITLE
how we handle feedback like "Not a Contact" ?

### DISCUSSION THREAD

**Discussion by ChocoPie02** [open-wa/wa-automate-nodejs#2615] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/65725993/161418139-8c3579bc-7f87-4c47-ad6d-c72f50782727.png)

just like this i use handling if else but nothing has found not either try and catch

```
> let e = await client.sendFile(nomer[0] + '@c.us','./file/Slip Gaji PT - ${akun[0]} - ${bulan[0]}.pdf', 'Slip Gaji PT  - ${akun[0]} - ${bulan[0]}.pdf', null, true);
>       if (e.length > 0) {
>         console.log('\x1b[41m%s\x1b[37m', e,'\x1b[0m');
>         fs.appendFileSync(logPath, '${e}|')
>       } else {
>         console.log('\x1b[42m%s\x1b[37m' , 'Sukses ${nomer[0]}@c.us |  ${akun[0]} | Slip Gaji PT  - ${akun[0]} - ${bulan}','\x1b[0m')
>         fs.appendFileSync(logPath, 'Sukses ${nomer[0]} |  ${akun[0]} | Slip Gaji PT  - ${akun[0]} - ${bulan}|')
>       }
```

**Comment by smashah** [comment]
@ChocoPie02 You need to have an existing chat with the contact [or get a "restricted" license](https://smashah.gumroad.com/l/open-wa?tier=1%20Restricted%20License%20Key) which will allow you to start chats with files.

I suggest not starting a chat with a file as it can increase your risk of being banned.

**Reply by ChocoPie02** [reply]
i already know about [restricted" license](https://smashah.gumroad.com/l/open-wa?tier=1%20Restricted%20License%20Key), what i ask is about how we handle the response "Not A contact", so i can filter it out from my list or add contact and start make initial chat

**Reply by ChocoPie02** [reply]
somehow, even using variable and check using if else its still going through else statement

**Comment by smashah** [comment, marked as answer]
@ChocoPie02 

Please track #2641

After that is resolved, you can set [`Config.OnError`](https://docs.openwa.dev/interfaces/api_model_config.ConfigObject.html#onError) to `"RETURN_ERROR"` to return an error which you can then handle.
